### PR TITLE
NMS-12753: Add encryption option for snmp config (#3054)

### PR DESCRIPTION
**Pull request by cgorantla**
* NMS-12753: Add encryption for snmp

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12753

### CODE REVIEW

**Comment by cgorantla** [comment]
Cherry picking this from features/space branch.
Review once again, Need feedback if it is okay to merge this into `foundation-2020`

**Review by RangerRick — CHANGES_REQUESTED** [review]
just marking as request changes for feedback on the changes to `apt-get` -- this looks suspect to me

**Review by RangerRick — APPROVED** [review]

**Comment by RangerRick** [comment]
@cgorantla looks like this needs some conflict-fixing

**Comment by mmahacek** [comment]
Docs path in 2021+ has changed

**Review by RangerRick — APPROVED** [review]

File: opennms-doc/guide-admin/src/asciidoc/index.adoc [hunk 1]
```diff
@@ -425,6 +425,7 @@ include::text/alarm-correlation/situation-feedback.adoc[]
 == Meta-Data
 include::text/meta-data.adoc[]
 
+
```
Comment (mmahacek) [inline opennms-doc/guide-admin/src/asciidoc/index.adoc:428]:
```suggestion
```

**Review by mmahacek — COMMENTED** [review]
As there will need to be manual merging into 2021 due to the Antora migration for docs, the fewer doc pages modified, the easier that merge will be.

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc [hunk 2]
```diff
@@ -90,6 +90,30 @@ We will then give an example of how to be more _directed_ and deliberate about y
 Automatically discovered entities are analyzed, persisted to the relational data store, and then managed based on the policies defined in the default foreign source definition.
 This is very similar to the way that entities were previously handled by the (now obsolete) Capsd daemon but with finer grained sense of control.
 
+==== Encrypting SNMP Credentials
```
Comment (Bonrob2) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc:93]:
```suggestion
==== Encrypt SNMP credentials
```

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc [hunk 3]
```diff
@@ -90,6 +90,30 @@ We will then give an example of how to be more _directed_ and deliberate about y
...
+
+You can save SNMP credentials like readCommunity/WriteCommunity, authPassPhrase/privPassPhrase on the file system in encrypted format.
```
Comment (Bonrob2) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc:95]:
```suggestion
You can save SNMP credentials like readCommunity/WriteCommunity or authPassPhrase/privPassPhrase on the file system in encrypted format.
```

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc [hunk 4]
```diff
@@ -90,6 +90,30 @@ We will then give an example of how to be more _directed_ and deliberate about y
...
+
+===== Encryption methodology
+
+A random password is generated (once) for `snmp-config` alias and it is stored in java keystore file at `$OPENNMS_HOME/etc/scv.jce`.
```
Comment (Bonrob2) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc:99]:
```suggestion
A random password is generated (once) for the `snmp-config` alias and it is stored in a Java keystore file at `$OPENNMS_HOME/etc/scv.jce`.
```

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc [hunk 5]
```diff
@@ -90,6 +90,30 @@ We will then give an example of how to be more _directed_ and deliberate about y
...
+This password is used to encrypt/decrypt SNMP credentials.
```
Comment (Bonrob2) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc:100]:
```suggestion
This password encrypts/decrypts SNMP credentials.
```
Does the password do this, or does one require the password so SNMP credentials can be encrypted/decrypted?

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc [hunk 6]
```diff
@@ -90,6 +90,30 @@ We will then give an example of how to be more _directed_ and deliberate about y
...
+
+===== Enabling Encrption on SNMP Config
```
Comment (Bonrob2) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc:102]:
```suggestion
===== Enable encryption on SNMP config
```

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc [hunk 7]
```diff
@@ -90,6 +90,30 @@ We will then give an example of how to be more _directed_ and deliberate about y
...
+
+Enable encryption on SNMP config by setting the following system property:
```
Comment (Bonrob2) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc:104]:
```suggestion
Set the following system property to enable encryption on SNMP config:
```

File: opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc [hunk 8]
```diff
@@ -90,6 +90,30 @@ We will then give an example of how to be more _directed_ and deliberate about y
...
+SNMP config that is modified through the user interface will be encrypted directly on the file system.
+Config that is updated on the file system needs to be saved with Karaf command `opennms-snmp:save-config` to get encrypted.
+
+NOTE: Currently, SNMP config encryption is not supported on the SNMP profiles.
+
+NOTE: Encryption Utils needs https://www.oracle.com/java/technologies/javase-jce-all-downloads.html[JCE] (Java Cryptography Extension) enabled on the JDK.
+
 ==== Automatic Discovery
 
 Currently in {opennms-product-name}, the ICMP is used to automatically provision node entities into {opennms-product-name}.
```
Comment (Bonrob2) [inline opennms-doc/guide-admin/src/asciidoc/text/provisioning/getting-started.adoc:119]:
```suggestion
Currently {opennms-product-name} uses the ICMP to automatically provision node entities into {opennms-product-name}.
```

**Review by Bonrob2 — COMMENTED** [review]
A few changes in the doc files.

**Comment by cgorantla** [comment]
Closing this for now. Will revisit when we have standard way of doing encryption for all the modules.

**Pull request closed without merging**